litellm/proxy/_types.py

```diff
@@ -1593,12 +1595,13 @@ class UserAPIKeyAuth(
     @model_validator(mode="before")
     @classmethod
     def check_api_key(cls, values):
-        if values.get("api_key") is not None:
-            values.update({"token": hash_token(values.get("api_key"))})
-            if isinstance(values.get("api_key"), str) and values.get(
-                "api_key"
-            ).startswith("sk-"):
-                values.update({"api_key": hash_token(values.get("api_key"))})
+        api_key = values.get("api_key")
```

Comment:
can you use `_hash_token_if_needed` in prox/utils.py here ?

Reply:
Hmmm, just tried that and it resulted in a circular import so it's not a trivial change.

Reply:
Well I think I can move `hash_token` and `_hash_token_if_needed` to `litellm.proxy._type` and then `utils.py` can have:

```python
from litellm.proxy._types import (
    hash_token,
    _hash_token_if_needed,
    ...
)
```

However, in order to test this with `make test-unit`, I will need #10662 to be merged...

Reply:
Well, I'll just commit and push the change (3b2587a6e) and see if CI passes...